exa/core/box.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2015-2020, Exa Analytics Development Team
+# Distributed under the terms of the Apache License 2.0
+"""
+Data container
+####################################
+An in-memory related data container.
+"""
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import networkx as nx
+from traitlets import Unicode, Dict
+
+import exa
+from exa import Base
+from exa.core.data import Data
+
+class Box(Base):
+    name = Unicode(help="string name of container")
+    meta = Dict(help="metadata dictionary")
+    description = Unicode(help="container description")
+
+    def copy(self, **kws):
+        return self.__class__(**kws)
+
+    def slice(self, key):
+        pass
+
+    def groupby(self):
+        pass
+
+    def info(self):
+        pass
+
+    def memory(self):
+        pass
+
+    def network(self):
+        pass
+
+    def save(self, name=None, directory=None, parquet_backend='pyarrow'):
+        """Save a bundle of files to a tarball."""
+        name = name or self.hexuid
+        directory = Path(directory) or exa.cfg.savedir
+        bundle = directory / name
+        bundle.mkdir(parents=True, exist_ok=True)
+        for data in self._data.values():
+            data.save(bundle)
+
+    def load(self):
+        pass
+
+    def from_hdf(cls):
+        # Do we opt for parquet+pickle?
```

Comment:
Can you say more about what you mean here? I like the use of parquet for flat column data (any types) and it plays well with `pandas.DataFrame` (etc.) objects. I think it might be better than the previous approach using `HDF5` files. For non-parquet-able data (like metadata) is there a way to include it in parquet files (I'm not aware of this possibility)? How might we handle situations like that for "standard" data that has "exa" metadata associated with it?

Reply:
I think most of this logic has been moved to the `save` methods on the `Data` and `Container` objects. It offers an approach for storing `pandas.DataFrame`s and `exa`'s associated metadata in 2 distinct file formats, compromising between complexity and convenience of use. 

We should probably consider how better to validate (or optionally enforce) the relationship between the 2 files, and perhaps even track a SHA sum of the parquet file tied to the `yaml` file. This would increase complexity and may limit flexibility of the API.